numpy/meson.build

```diff
@@ -10,15 +10,20 @@ endif
 
 # Platform detection
 is_windows = host_machine.system() == 'windows'
-is_mingw = is_windows and cc.get_id() == 'gcc'
+is_mingw = is_windows and cc.get_id() in ['gcc', 'clang']
```

Comment:
This can't be right I think, it will break using regular Clang with MSVC-built CPython. Conda-forge uses that for SciPy since recently, so it's not a "no one will do that" situation.

This check is for "this is a Mingw" compiler, so I don't quite understand how you'd get Clang here?

Reply:
Do you mean `clang-cl`? `clang` on Windows means clang providing a gcc interface, I only know mingw providing that: https://mesonbuild.com/Reference-tables.html#compiler-ids

Reply:
@h-vetinari conda-forge does have a `clang.exe` with GCC syntax in addition to the main `clang-cl.exe`, right?

Reply:
Yes, we have both

Reply:
And does clang.exe with GCC syntax use mingw-w64?

Reply:
I had a look for the answer in conda-forge feedstocks, but could find it so quickly. https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/main/recipe/meta.yaml#L19 skips Windows.

Reply:
The recipe for clang on windows is in a different [place](https://github.com/conda-forge/clang-win-activation-feedstock/).

> And does clang.exe with GCC syntax use mingw-w64?

No, mingw isn't necessary to run clang.

Reply:
I see. I just installed clang from upstream and played around with meson, and it indeed is as you said. I for some reason never encountered this combination. I'll guess I'll best check for `__MINGW32__ ` then. Thanks!

Reply:
I've added a check for `__MINGW32__` instead.

Reply:
What does cc.get_id() show for mingw? 

FWIW: the intel oneAPI compiler is also a clang-cl based compiler It returns a intel specific string for `get_id()`: `intel-llvm-cl`

Reply:
for gcc it returns 'gcc' and clang it returns 'clang'

Reply:
> I've added a check for `__MINGW32__` instead.

Then I guess this is the best that can be done. Is there a `__MINGW64__` as well?

Reply:
> __MINGW64__

yes, that's defined for 64 only, while `__MINGW32__` is defined for 32 and 64 bit, same as `_WIN32` and `_WIN64`